src/api/core/warren/models.py

```diff
@@ -39,9 +43,47 @@ class DailyCount(BaseModel):
     date: Date
     count: int = 0
 
+    def __add__(self, other):
+        """Add counters for two DailyCount instances with the same date."""
+        if self.date != other.date:
+            raise Exception("Cannot add DailyCounts with different dates")
+        return DailyCount(date=self.date, count=self.count + other.count)
+
+    def __radd__(self, other):
+        """Add counters for two DailyCount instances with the same date."""
+        if self.date != other.date:
+            raise Exception("Cannot add DailyCounts with different dates")
+        return DailyCount(date=self.date, count=self.count + other.count)
+
 
 class DailyCounts(BaseModel):
     """Base model to represent daily counts summary."""
 
     total_count: int = 0
     count_by_date: List[DailyCount] = []
+
+    @classmethod
+    def from_range(cls, date_range: DatetimeRange):
+        """Initialize DailyCounts from a date range."""
+        return cls(
+            count_by_date=[
+                DailyCount(date=d)
+                for d in arrow.Arrow.range("day", date_range.since, date_range.until)
+            ]
+        )
+
+    def merge(self, counts: List[DailyCount]):
```

Comment:
It wouldn't change that much, but I find it more intuitive to merge two objects instead of an object and a specific field of another instance of the same object type. WDYT?

```suggestion
    def merge(self, other: DailyCounts):
```

Reply:
Initially I tried the following, however while using it in the `compute` method, I figured out I had to instantiate a `DailyCounts` object just to pass a List `DailyCount`. Having two methods when we need it could be great? I could rename it to `merge_counts`, WDYT? 

```
    def merge(self, other: DailyCounts):
        return self.merge_counts(other.counts)

    def merge_counts(self, other: List[DailyCount]):
    ...
    
```

Reply:
I see. That would be an acceptable solution!